src/Hint/Monad.hs

```diff
@@ -110,6 +147,48 @@ monadHint _ _ d = concatMap (f Nothing Nothing) $ childrenBi d
         isHsDo (L _ HsDo{}) = True
```

Comment:
this is available from ghc-lib-parser-ex as `Language.Haskell.GhclibParserEx.GHC.Hs.Expr.isDo`